src/SSHLibrary/abstractclient.py

```diff
@@ -346,7 +348,7 @@ def read_until(self, expected):
         :returns: The read output, including the encountered `expected` text.
         """
         expected = self._encode(expected)
-        return self._read_until(lambda s: expected in s, expected)
+        return self._read_until(lambda s: expected in s.encode(self.config.encoding), expected)
```

Comment:
Instead of encoding `s` (i.e. output that is read), it ought be better to *not* encode `expected`. The `_read_until` method used `read_char` which always returns Unicode and we can expect the given `expected` to be Unicode as well. Noe encoding `expected` would also avoid the need to decode it in `_read_until` when constructing an error message.